### PR TITLE
Pin DaCe to PyPI supported version 0.13

### PR DESCRIPTION
**Pull request by gronerl**
Update dace version to v0.13 (which can also be found in PyPI
Some of the required changes trigger symbol errors in compilation, so I added already some things from the fv3core works (specialization of transient strides)

Solves #686

### CODE REVIEW

**Comment by gronerl** [comment]
bors try

**Review by havogt — APPROVED** [review]
As I don't know the details of these changes, just some comments for you to double-check that everything makes sense.

File: src/gtc/passes/oir_dace_optimizations/horizontal_execution_merging.py [hunk 1]
```diff
@@ -241,10 +240,10 @@ def _merge_source_locations(
             dinfo.filename = dinfo.filename or right.debuginfo.filename
         return dinfo
 
-    def apply(self, sdfg: dace.SDFG) -> None:
+    def apply(self, state, sdfg: dace.SDFG) -> None:
         state = sdfg.node(self.state_id)
```
[the remark this answers is not in the document]
Comment (gronerl) [reply src/gtc/passes/oir_dace_optimizations/horizontal_execution_merging.py]: yes, it's redundant

File: src/gtc/dace/utils.py [hunk 2]
```diff
@@ -84,6 +85,22 @@ def array_dimensions(array: dace.data.Array):
     return dims
 
 
+def replace_strides(arrays, get_layout_map):
```
[the remark this answers is not in the document]
Comment (gronerl) [reply src/gtc/dace/utils.py:88]: trust me, bro

**Comment by gronerl** [comment]
bors try

**Pull request merged**